x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -180,26 +179,35 @@ export function setSelectedLayer(layerId) {
   };
 }
 
-export function updateLayerOrder(newLayerOrder) {
-  return {
-    type: UPDATE_LAYER_ORDER,
-    newLayerOrder
+export function removeTransientLayer() {
+  return async (dispatch, getState) => {
+    const transientLayerId = getTransientLayerId(getState());
+    if (transientLayerId) {
+      dispatch(removeLayer(transientLayerId));
+      dispatch(setTransientLayer(null));
```

Comment:
Precede these two with `await`. Callers of this action-creator can then be guaranteed that this related state is cleaned up. (e.g. two independent actions need to resolve. The clearing of the transient-property and the removal of the layer. Both need to have occurred, otherwise the store is in an invalid state).